src/SharedInfrastructure/InliningOptions.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+// Uncomment this for verbose profiler results. DO NOT PUSH TO MAIN!
+// #define PROFILING
+using System.Runtime.CompilerServices;
+
+namespace SixLabors
+{
+    /// <summary>
+    /// Global inlining options. Helps temporarily disable inlining for better profiler output.
+    /// </summary>
+    internal static class InliningOptions
+    {
+#if PROFILING
+        public const MethodImplOptions HotPath = MethodImplOptions.NoInlining;
+        public const MethodImplOptions ShortMethod = MethodImplOptions.NoInlining;
+#else
+    #if SUPPORTS_HOTPATH
+        public const MethodImplOptions HotPath = MethodImplOptions.AggressiveOptimization;
```

Comment:
@JimBobSquarePants Possibly dumb quesiton, why are we just using `MethodImplOptions.AggressiveOptimization` in this case instead of `MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization`? I've seen them used at the same time quote often around CoreCLR, and I don't think that the optimization flag implies inlining as well - don't we want both those JIT perks for hot paths? 🤔

Reply:
I've only seen them used as flags 

`MethodImplOptions.AggressiveOptimization | MethodImplOptions.AggressiveInlining`

I haven't managed to find any documentation on the new attribute at all, nor could I find the issue that introduced it so I thought I better keep them separate as the original enum defines.

Reply:
Maybe we should combine them. @antonfirsov Do you have any insight?

Reply:
Fond the original CoreFX issue. 

https://github.com/dotnet/runtime/issues/27370

Dunno why I couldn't before.

Reply:
@Sergio0694 Looks like you definitely want to keep them separate. Relevant conversation starts here.

https://github.com/dotnet/runtime/issues/27370#issuecomment-420439226

Reply:
In general we should have a separate flag for `AggressiveOptimization`, haven't noticed that it exists until recently.

However, let's not add `InliningOptions` to SharedInfrastructure for now, because keeping it in this separate repo will remove all our flexibility for expermientats.
`InliningOptions` is mostly here to support having more verbose profiler results when we want. I need more time to figure out what is the best approach for `AggressiveOptimization`, so let's keep these concerns separate from each other.

All `Guard` methods should be simply `MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization` instead. It's better to have them always inlined, even when we do profiling.

Reply:
@antonfirsov  No worries, will revert for now. I'll add to to ImageSharp main in a separate PR.

@Sergio0694 If you let me know what the expected message should be in the ThrowArgumentOutOfRangeException and I'll get it updated.

Reply:
@antonfirsov wait, I'm confused. Why would you want the `AggressiveOptimization` flag? According to the conversation in those linked issues, one of it's main advantages is that it tends to inline method called _from_ there, but we don't need that at all since all the throw methods are marked as `NoInlining` anyway. We only care about those `Guard` APIs always being inlined, and from what I've read there's a risk that adding `AggressiveOptimization` might conflict with that. Unless I'm wrong on this point, which could be too 🤔

Reply:
@JimBobSquarePants I'm not sure what you mean there, what exception message? I mean, I can see the throw methods for the out of range errors with a formatted message already, what are you referring to here?

Reply:
@Sergio0694 I suppose inlining is not the only magic provided by `AggressiveOptimization`, but you're right `Guard` methods are so short, there really isn't anything to agressively optimize, and if you say it might bring other risks, let's leave it out for now!

Reply:
I'm not 100% sure since that conversation was really long and I could just read up to some point, but apparently there were a lot of various implications depending on the scenario. The main takeaway I got was this:

• `AggressiveInlining` just pushes the JIT to inline the (marked) callee into its caller
• `AggressiveOptimization` pushes the JIT to try to inline callees from their (marked) caller (the one with the attribute). It also suggests to go straight to the best JIT tiering possible.

In out case, the `Guard` APIs don't call functions that could benefit from inlinig, in fact all the helpers should _not_ be inlined, and are marked as such. We only care about those `Guard` APIs themselves being inlined into _their_ callers, so yeah I'd say (at least for now) the safe bet would be to just use `AggressiveInlining` 👍

Reply:
@Sergio0694 What I mean is that the unit tests expect a different value for the message than the  actual output. Shall I just update the test to match?

https://github.com/SixLabors/SharedInfrastructure/pull/7/files#diff-f73475a089d950d0e0a672480a854ebcR45

Reply:
Ah my bad, I didn't consider the fact the tests were also matching the exact error message as well. Sure thing, yeah that would be perfect then 👍